src/dopri.c

```diff
@@ -805,7 +801,56 @@ void ylag_vec_int(double t, const int *idx, size_t nidx, double *y) {
   }
 }
 
+
+void dopri_eval(dopri_data *obj, double t, double *y, double *dydt) {
+  dopri_print_eval(obj, t, y);
+  obj->target(obj->n, t, y, dydt, obj->data);
+  obj->n_eval++;
+}
+
+
+void dopri_print_step(dopri_data *obj, double h) {
+  if (obj->verbose >= VERBOSE_STEP) {
+    if (obj->callback == R_NilValue) {
+      Rprintf("[step] t: %f, h: %e\n", obj->t, h);
+    } else {
+      dopri_callback(obj, obj->t, h, obj->y);
+    }
+  }
+}
+
+
+void dopri_print_eval(dopri_data *obj, double t, double *y) {
+  if (obj->verbose >= VERBOSE_EVAL) {
+    if (obj->callback == R_NilValue) {
+      Rprintf("[eval] t: %f\n", t);
+    } else {
+      dopri_callback(obj, obj->t, NA_REAL, obj->y);
+    }
+  }
+}
+
+
+void dopri_callback(dopri_data *obj, double t, double h, double *y) {
+  SEXP callback = VECTOR_ELT(obj->callback, 0);
+  SEXP env = VECTOR_ELT(obj->callback, 1);
+
+  SEXP r_t = PROTECT(ScalarReal(obj->t));
+  SEXP r_h = PROTECT(ScalarReal(h));
+  SEXP r_y = PROTECT(allocVector(REALSXP, obj->n));
```

Comment:
Is there a reason here that `t` and `*y` are passed in but are then not used in method?

Reply:
yeah, this is an error - fixing it up now and adding a better test